mne/channels/channels.py

```diff
@@ -1536,3 +1539,150 @@ def make_1020_channel_selections(info, midline="z"):
+        shape ``(n_times)``, ``(n_epochs, n_times)``, or ``(n_evokeds,
+        n_times)``. For example with an instance of Raw::
+
+            method = lambda data: np.median(data, axis=0)
+
+        Defaults to ``mean``.
+    keep_stim : bool
+        If True (default False), keep stimulus channels.
+    drop_bad : bool
+        If True (default False), drop channels marked as bad.
+
+    Returns
+    -------
+    combined_inst : instance of Raw, Epochs, or Evoked
+        An MNE-Python object with channels combined for each group defined in
+        the ``groups`` parameter.
+    """
+    from ..io import BaseRaw, RawArray
+    from ..epochs import BaseEpochs
+    from ..evoked import Evoked
+    from .. import EpochsArray, EvokedArray
+
+    # Check to see if data is preloaded
+    _check_preload(instance, 'Combining channels')
+
+    ch_axis = len(instance._data.shape) - 2  # idx of the channel axis
+    ch_idx = list(range(instance.info['nchan']))
+    ch_types = _get_channel_types(instance.info)
+    groups = OrderedDict(deepcopy(groups))
+
+    # Convert string values of ``method`` into callables
+    # XXX Possibly de-duplicate with _make_combine_callable of mne/viz/utils.py
+    if isinstance(method, str):
+        method_dict = {key: partial(getattr(np, key), axis=ch_axis)
+                       for key in ('mean', 'median', 'std')}
+        try:
+            method = method_dict[method]
+        except KeyError:
+            raise ValueError('"method" must be a callable, or one of '
+                             f'"mean", "median", or "std"; got "{method}".')
+
+    # Create new instance
+    if not isinstance(keep_stim, bool):
+        raise TypeError('"keep_stim" must be of type bool, not '
+                        f'{type(keep_stim)}.')
+    if keep_stim is True:
+        try:
+            combined_inst = instance.copy().pick_types(meg=False, stim=True)
+            # Create clean info to avoid inconsistencies
+            stim_ch_names = combined_inst.info['ch_names']
+            stim_ch_types = _get_channel_types(combined_inst.info)
+            combined_inst.info = create_info(sfreq=instance.info['sfreq'],
+                                             ch_names=stim_ch_names,
+                                             ch_types=stim_ch_types)
+        except ValueError:
+            raise ValueError('Could not find stimulus channels.')
+    else:
+        combined_inst = None
+
+    # Get indices of bad channels
+    if not isinstance(drop_bad, bool):
+        raise TypeError('"drop_bad" must be of type bool, not '
+                        f'{type(drop_bad)}.')
+    if drop_bad is False:
+        ch_idx_bad = []
+    elif not instance.info['bads']:
+        ch_idx_bad = []
+    else:
+        ch_idx_bad = pick_channels(instance.info['ch_names'],
+                                   instance.info['bads'])
+
+    # Check correctness of combinations
+    for this_group, these_picks in groups.items():
```

Comment:
```suggestion
    for this_group, this_picks in groups.items():
```
is more common idiom.